coderd/database/queries/workspaces.sql

```diff
@@ -687,3 +687,39 @@ UPDATE workspaces SET favorite = true WHERE id = @id;
 
 -- name: UnfavoriteWorkspace :exec
 UPDATE workspaces SET favorite = false WHERE id = @id;
+
+-- name: GetWorkspacesAndAgentsByOwnerID :many
+SELECT
+	workspaces.id as id,
+	workspaces.name as name,
+	job_status,
+	transition,
+	(array_agg(ROW(agent_id, agent_name)::agent_id_name_pair) FILTER (WHERE agent_id IS NOT NULL))::agent_id_name_pair[] as agents
```

Comment:
Isn't there a generic type we can use? Adding a new type to the schema for this one query seems like a lot.  Does it end up being super gross on the Go side or something?

Reply:
`sqlc` just spits out `interface{}`, as it doesn't support generating composite types at all: https://github.com/sqlc-dev/sqlc/issues/2760